biz.aQute.repository/src/aQute/bnd/deployer/repository/wrapper/InfoRepositoryWrapper.java

```diff
@@ -168,21 +168,21 @@ public void findProviders(Map<Requirement,List<Capability>> result, Collection<
 			throws Exception {
 		init();
 
-		nextReq: for (Requirement req : requirements) {
+		for (Requirement req : requirements) {
+			result.putIfAbsent(req, new ArrayList<Capability>(1));
```

Comment:
FYI - This is `putIfAbsent()` instead of `put()` because I don't know that we are guaranteed to have unique `Requirement`s being passed in.